crates/sui-core/src/execution_scheduler/execution_scheduler_impl.rs

```diff
@@ -207,48 +226,71 @@ impl ExecutionSchedulerAPI for ExecutionScheduler {
         certs: Vec<(Schedulable, ExecutionEnv)>,
         epoch_store: &Arc<AuthorityPerEpochStore>,
     ) {
-        // schedule all transactions immediately
         let mut txns = Vec::with_capacity(certs.len());
-        let mut rest = Vec::new();
+        let mut randomness_txns = Vec::new();
+        let mut settlement_txns = Vec::new();
 
         for (schedulable, env) in certs {
-            if let Schedulable::Transaction(tx) = schedulable {
-                txns.push((tx, env));
-            } else {
-                rest.push((schedulable, env));
+            match schedulable {
+                Schedulable::Transaction(tx) => {
+                    txns.push((tx, env));
+                }
+                Schedulable::RandomnessStateUpdate(_, _) => {
+                    randomness_txns.push((schedulable.key(), env));
+                }
+                Schedulable::AccumulatorSettlement(_, _) => {
```

Comment:
Is there a reason to schedule this from consensus in the first place?
It seems that you could just not send it through from consensus handler to scheduler, and let the checkpoint builder enqueue it?

Reply:
I was just hoping to avoid yet another special case in consensus handler where we have a transaction that is added to checkpoint roots, but not enqueued for execution (or vice versa). if we ever have time to clean up that code, it would be good to remove more of those special cases

Reply:
I see.
Maybe we don't have to specialize it in consensus handler? i.e. we send it over to the scheduler, but scheduler can ignore it if it has a way to know this is from consensus

Reply:
The reason I am against the current approach is that we are introducing a lot of complexity just because this transaction cannot be scheduled directly from consensus handler, and must wait for checkpoint builder. There is really no other dependency. So we might as well just schedule it from checkpoint builder.

Reply:
Good point. Let me think about it a bit more.

Reply:
What about just adding the assigned accumulator object version number to the Settlement Schedulable? That way, checkpoint builder can easily figure out the ExecEnv which here is really just about the version. I think that will still be simpler and easier to reason about then scheduling it twice.